lib/queue.js

```diff
@@ -61,6 +63,19 @@ var Queue = function Queue(name, redisPort, redisHost, redisOptions){
     redisOptions.db = redisOpts.DB;
   }
 
+  if (_.isString(redisPort)) {
+    try {
+      var redisUrl = url.parse(redisPort);
+      assert(_.isObject(redisHost) || _.isUndefined(redisHost));
+      redisOptions =  redisHost || {};
+      redisPort = redisUrl.port;
+      redisHost = redisUrl.hostname;
+      redisOptions.password = redisUrl.auth? redisUrl.auth.split(':')[1]: void 0;
+    } catch (e) {
+      throw new Error('Malformed connection string');
```

Comment:
maybe useful to log the err thrown by the try so that the user can get more information about why the connection string is malformed.